Tests/NIOCoreTests/NIODecodedAsyncSequenceTests.swift

```diff
@@ -125,4 +110,82 @@ struct NIODecodedAsyncSequenceTests {
             }
         }
     }
+
+    @Test(arguments: SplittingTextTestArgument.all)
+    private func splittingTextWorksSimilarToStandardLibraryStringSplitFunctions(
+        argument: SplittingTextTestArgument
+    ) async throws {
```

Comment:
Test against `String.split`. This allows for a bigger test suite with us writing much less code and not having to hard-code the end result.
I _could_ also add a test with some hardcoded results. Then we'll also have this test as the "more extensive" test.

Reply:
These are the tests after modification to use the verifier: https://github.com/apple/swift-nio/pull/3411/commits/765458779cde9db00a78c2855636b77d2766e2d1

I've reverted that commit for now.